zerver/views/registration.py

```diff
@@ -539,38 +539,33 @@ def find_account(request: HttpRequest) -> HttpResponse:
     url = reverse('zerver.views.registration.find_account')
 
     emails = []  # type: List[str]
+    emails_sent = []  # type: List[str]
+    emails_ignore = []  # tyep: List[str]
     if request.method == 'POST':
         form = FindMyTeamForm(request.POST)
         if form.is_valid():
             emails = form.cleaned_data['emails']
-            for user in UserProfile.objects.filter(
-                    delivery_email__in=emails, is_active=True, is_bot=False,
-                    realm__deactivated=False):
-                context = common_context(user)
-                context.update({
-                    'email': user.delivery_email,
-                })
-                send_email('zerver/emails/find_team', to_user_ids=[user.id], context=context,
+            for user in UserProfile.objects.filter(delivery_email__in=emails):
+                if user.is_active is True and user.is_bot is False and user.realm.deactivated is False:
+                    context = common_context(user)
+                    context.update({
+                        'email': user.delivery_email,
+                    })
+                    send_email('zerver/emails/find_team', to_user_ids=[user.id], context=context,
+                               from_address=FromAddress.SUPPORT)
+                    if user.delivery_email not in emails_sent:
+                        emails_sent.append(user.delivery_email)
+                else:
+                    emails_ignore.append(user.delivery_email)
+            for email in list((set(emails).difference(set(emails_sent))).difference(set(emails_ignore))):
```

Comment:
I don't understand what's happening with this `.difference` logic; can you write it a different way that's more clear?

Reply:
We want to send emails to only the users and non-users. So we find the difference between the set of emails received, emails sent in first iteration (registered user) and bot emails.
By doing this we are making sure we're only sending mails to non-registered users in second iteration.

Right now the logic is ignoring users of deactivated realms and inactive users same as bot users (we are ignoring them and not sending emails). I guess we should still send them emails saying "You are not registered to any Zulip org" or do we want to ignore them?